File: backend/models/ChatHistory.js
const { DataTypes } = require("sequelize");
const sequelize = require("../config/db");

const ChatHistory = sequelize.define(
  "ChatHistory",
  {
    id: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: true,
    },
    role: {
      type: DataTypes.STRING(10),
      allowNull: false,
    },
    text: {
      type: DataTypes.TEXT,
      allowNull: false,
    },
  },
  {
    tableName: "ChatHistory",
    timestamps: true, // Bắt buộc để true để dùng createdAt/updatedAt
  },
);

module.exports = ChatHistory;

File: backend-csharp/Models/ChatHistory.cs
using System;
using System.Collections.Generic;

namespace WebFashion.Api.Models;

public partial class ChatHistory
{
    public int Id { get; set; }

    public string Role { get; set; } = null!;

    public string Text { get; set; } = null!;

    public DateTimeOffset CreatedAt { get; set; }

    public DateTimeOffset UpdatedAt { get; set; }
}
